g2p_seq2seq/app.py

```diff
@@ -54,20 +55,29 @@
 tf.app.flags.DEFINE_integer("max_steps", 0,
                             "How many training steps to do until stop training"
                             " (0: no limit).")
+tf.app.flags.DEFINE_boolean("reinit", False,
+                            "Set to True for training from scratch.")
 
 FLAGS = tf.app.flags.FLAGS
 
 def main(_=[]):
   """Main function.
   """
-  if FLAGS.train:
-    with tf.Graph().as_default():
-      g2p_model = G2PModel(FLAGS.model)
+  with tf.Graph().as_default():
+    if FLAGS.train:
+      g2p_model = G2PModel(FLAGS.model, train_flag=True)
```

Comment:
Not much sense for such constructor here, it returns immediately because train_flag is True. It might have sense to use smaller constructor and separate a method `load` to use in training and decoding.